plugins/modules/foreman_domain.py

```diff
@@ -31,10 +31,14 @@
   name:
     description: The full DNS domain name
     required: true
-  dns_proxy:
+  dns:
```

Comment:
I actually liked the dns_proxy name more because it was more explicit.

Reply:
>     * we use `from_ip` which does sound nicer, but my brain likes to have our and Foremans params to match (if possible)

We can not have it both. And there is still the alias.

Reply:
This is tricky because you don't want to literally use `dns_id`. Because we already do some conversion on top (by retrieving the actual object and getting the ID) maybe this is fine?

Reply:
`dns_id` is not visible to ansible (not even mentioned in the `argument_spec`). It is _only_ used internally towards the api.

Reply:
Let me try to be a bit less brief :) We don't want to use the literal API default, I think we agree on that. Then the question becomes what we should use. Just reading `dns` without context makes me think it's a boolean (manage or not) while with `dns_proxy` it's clearer to me that I need to specify a Foreman Proxy with the DNS feature. That's why I liked the `dns_proxy` option here. Can we use `dns_proxy` as default and `dns` as alias perhaps?

Reply:
I think, we can even use dns_proxy (which i prefer, too) and not even provide dns as an alias as before. But we'd need something like https://github.com/theforeman/foreman-ansible-modules/pull/376#discussion_r308188471 again.
I'd like @evgeni 's thought before flip-flopping that one.

Reply:
See below (thanks fasthub!) and tl;dr: dns_proxy, no alias is fine here for ne

Reply:
In the current implementation (i.e. without `alt_names`), i think, this is not possible, as the `flat_name` _must _ be `dns_id` and we _need_ to have `dns` in the generated `entity_spec` to correctly translate what foreman delivers in the `find_resource`.

Reply:
There is one more option if we supported both names: We could lie in the docs about which one is the alias.

Reply:
Uh, and how does it work now with dns_proxy?

Reply:
Atm dns_proxy is the alias.

Reply:
No, not in https://github.com/theforeman/foreman-ansible-modules/blob/827eb5a8629c41a7e70311e08ff1ea967b258433/plugins/modules/foreman_domain.py#L92 ;)

Reply:
Ah, i meant in this pr.
And honestly, i think in the current master it is actually broken: See #360

Reply:
But the API response to a PUT/POST or GET will include both, `dns_id` and `dns` 
https://github.com/theforeman/foreman-ansible-modules/blob/827eb5a8629c41a7e70311e08ff1ea967b258433/tests/test_playbooks/fixtures/domain-1.yml#L43-L48

so technically we don't need to "flatten" `dns` into `dns_id` as that's already there?

Reply:
I conclude, that means we don't need it here, and therefore we can add it later if needed.

Reply:
So @mdellweg pointed out to me on IRC that while this is true for the domain→dns relationship, it's not true for organization→media and organization→templates. However, we don't diverge from the API name in the Ansible spec there so the issue does not arise.

We could add an `api_name` to the spec in the future to avoid that.

Bit *I* think doing this in the future is sufficient.